File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        int mod=1000000007;
        int even=1;
        int odd=0;
        int res=0;
        int prefix=0;
        for(int num:arr){
            prefix=prefix+num;
            if(prefix%2==0){
                res=res+odd;
                even++;
            }else{
                res=res+even;
                odd++;
            }
            res=res % mod;
        }
        return res;
    }
}

File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr) {
        int n=arr.size();
        unordered_map<int,int>mp;
        int mod=1e9+7;
        mp[0]=1;
        int sum=0;
        int ans=0;
        for(int i=0;i<n;i++){
            sum+=arr[i];
            if(sum%2==0){
                if(mp.find(1)!=mp.end())
                ans=(ans%mod+mp[1]%mod)%mod;
                mp[0]++;            
        }else{
            if(mp.find(0)!=mp.end())
            ans=(ans%mod+mp[0]%mod)%mod;
            mp[1]++;
        }
    }
    return ans;
    }

};